spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/elasticsearch/sniff/ElasticsearchSnifferProperties.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.elasticsearch.sniff;
+
+import org.elasticsearch.client.sniff.NodesSniffer;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Configuration properties for Elasticsearch sniffer.
+ *
+ */
+@ConfigurationProperties(prefix = "spring.elasticsearch.sniff")
+public class ElasticsearchSnifferProperties {
+
+	/**
+	 * Sniffer interval millis
+	 */
+	private Duration sniffIntervalMillis = Duration.ofMillis(1L);
```

Comment:
A `Duration` property should not have `mills` (or any unit) in its name since users are  free to chose a different unit if they want to.

Also, the default value is wrong, it should be 5 min.